code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,344 @@
+/obj/machinery/ftl_shunt
+	anchored = 1
+	icon = 'icons/obj/shunt_drive.dmi'
+
+/obj/machinery/ftl_shunt/core
+	name = "FTL shunt core"
+	desc = "An immensely powerful particle accelerator capable of forming a micro-wormhole and shunting an entire ship through it in a nanosecond."
+
+	var/list/fuel_ports = list() //We mainly use fusion fuels.
+	var/charge_time //We don't actually charge to a certain amount of power, we just charge for x amount of time depending on host ship mass.
+	var/charge_started
+	var/charging = FALSE
+	var/shunt_x = 1
+	var/shunt_y = 1
+	var/obj/machinery/computer/ship/ftl/ftl_computer
+	var/required_fuel_joules
+	var/cooldown_delay = 5 MINUTES
+	var/cooldown
+	var/time_multiplier = 5 //The multiplier on how long it takes the shunt drive to charge.
+	var/max_jump_distance = 8 //How many overmap tiles can this move the ship?
+
+	var/static/datum/announcement/priority/ftl_announcement = new(do_log = 0, do_newscast = 1, new_sound = sound('sound/misc/notice2.ogg'))
+
+	var/shunt_start_text = "Attention! All hands brace for faster-than-light transition! ETA: %%TIME%%"
+	var/shunt_cancel_text = "Attention! Faster-than-light transition cancelled."
+	var/shunt_complete_text = "Attention! Faster-than-light transition completed."
+
+	use_power = POWER_USE_OFF
+	power_channel = EQUIP
+	idle_power_usage = 1600
+	active_power_usage = 150000
+	icon_state = "bsd"
+
+/obj/machinery/ftl_shunt/core/Initialize()
+	. = ..()
+	find_ports()
+
+/obj/machinery/ftl_shunt/core/Destroy()
+	. = ..()
+	for(var/obj/machinery/ftl_shunt/fuel_port/FP in fuel_ports)
+		FP.master = null
+		fuel_ports -= FP
+	ftl_computer.linked_core = null
+	ftl_computer = null
+
+/obj/machinery/ftl_shunt/core/proc/find_ports()
+	for(var/obj/machinery/ftl_shunt/fuel_port/FP in range(7))
```

Comment:
Consider using a local machine network extension like the RUST

Reply:
Outside of my technical understanding.